theories/distributions/Mu_mem.ec

```diff
@@ -87,3 +87,37 @@ proof.
   rewrite cardE -(perm_eq_size (undup l)) 1:oflistK //.
   by rewrite ler_wpmul2r // le_fromint size_undup.
 qed.
+
+(* This is more general than the mu_mem lemmas for fset *)
+lemma mu_mem_le_fsize (m : ('a, 'b) fmap) (d : 'c distr) p bd :
+  (forall u, u \in m => mu d (p u) <= bd) =>
+  mu d (fun r => exists u, u \in m /\ p u r) <= (fsize m)%r * bd.
+proof.
+elim/fmapW : m; first by rewrite mu0_false; smt(mem_empty fsize_empty).
+move => m k v fresh_k IH H.
+rewrite (@mu_eq _ _ (predU (p k)
+  (fun (r : 'c) => exists (u : 'a), (u \in m) /\ p u r))); 1: smt(mem_set).
+rewrite mu_or; apply ler_naddr; 1: smt(mu_bounded).
+by rewrite fsize_set -mem_fdom fresh_k fromintD mulrDl; smt(mem_set).
+qed.
+
+(* this would be the corresponding generalization for fset *)
+lemma mu_mem_le_gen (s:'a fset) (d:'b distr) p (bd:real) : 
+  (forall (x : 'a), mem s x => mu d (p x) <= bd) =>
+  mu d (fun r => exists x, x \in s /\ p x r) <= (card s)%r * bd.
```

Comment:
This probably should not stay in `SmtMap` as it doesn't even mention maps. So don't merge this as is. ^^

Reply:
That's not in `SmtMap`, though...

Reply:
Fair enough. :roll_eyes: 

Still, the question is whether to push the generalization through to `Mu_mem` (see the aborted alternative proof of `mu_mem_le`. Presumably, the same would apply to the other variants of `mu_mem`.